warehouse/migrations/versions/0cb51a600b59_rename_provider_to_publisher.py

```diff
@@ -0,0 +1,301 @@
+            name="_github_oidc_publisher_uc",
+        ),
+    )
+    op.create_table(
+        "oidc_publisher_project_association",
+        sa.Column(
+            "id",
+            postgresql.UUID(as_uuid=True),
+            server_default=sa.text("gen_random_uuid()"),
+            nullable=False,
+        ),
+        sa.Column("oidc_publisher_id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.Column("project_id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["oidc_publisher_id"],
+            ["oidc_publishers.id"],
+        ),
+        sa.ForeignKeyConstraint(
+            ["project_id"],
+            ["projects.id"],
+        ),
+        sa.PrimaryKeyConstraint("id", "oidc_publisher_id", "project_id"),
+    )
+    op.create_table(
+        "pending_oidc_publishers",
+        sa.Column(
+            "id",
+            postgresql.UUID(as_uuid=True),
+            server_default=sa.text("gen_random_uuid()"),
+            nullable=False,
+        ),
+        sa.Column("discriminator", sa.String(), nullable=True),
+        sa.Column("project_name", sa.String(), nullable=False),
+        sa.Column("added_by_id", postgresql.UUID(as_uuid=True), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["added_by_id"],
+            ["users.id"],
+        ),
+        sa.PrimaryKeyConstraint("id"),
+    )
+    op.create_table(
+        "pending_github_oidc_publishers",
+        sa.Column("repository_name", sa.String(), nullable=True),
+        sa.Column("repository_owner", sa.String(), nullable=True),
+        sa.Column("repository_owner_id", sa.String(), nullable=True),
+        sa.Column("workflow_filename", sa.String(), nullable=True),
+        sa.Column("id", postgresql.UUID(as_uuid=True), nullable=False),
+        sa.ForeignKeyConstraint(
+            ["id"],
+            ["pending_oidc_publishers.id"],
+        ),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint(
+            "repository_name",
+            "repository_owner",
+            "workflow_filename",
+            name="_pending_github_oidc_publisher_uc",
+        ),
+    )
+    op.create_index(
+        op.f("ix_pending_oidc_publishers_added_by_id"),
+        "pending_oidc_publishers",
+        ["added_by_id"],
+        unique=False,
+    )
+    op.add_column(
+        "macaroons",
+        sa.Column("oidc_publisher_id", postgresql.UUID(as_uuid=True), nullable=True),
+    )
+    op.create_index(
+        op.f("ix_macaroons_oidc_publisher_id"),
+        "macaroons",
+        ["oidc_publisher_id"],
+        unique=False,
+    )
+    op.create_foreign_key(
+        None, "macaroons", "oidc_publishers", ["oidc_publisher_id"], ["id"]
+    )
+
+    # Macaroon -> (User XOR OIDCPublisher)
+    op.create_check_constraint(
+        "_user_xor_oidc_publisher_macaroon",
+        table_name="macaroons",
+        condition="(user_id::text IS NULL) <> (oidc_publisher_id::text IS NULL)",
+    )
```

Comment:
Manually added this because it was not picked up by alembic automatically.

Reply:
Looks correct to me! Thanks for catching that.